components/aws/sagemaker/commonv2/sagemaker_component.py

```diff
@@ -689,3 +831,24 @@ def _write_output(
 
         Path(output_path).parent.mkdir(parents=True, exist_ok=True)
         Path(output_path).write_text(write_value)
+
+    def _is_upgrade(self):
+        """If the resource already exists the component assumes that the user wants to upgrade
+        Returns:
+            Bool: If the resource is being upgraded or not.
+        Raises:
+            Exception
+        """
+        try:
+            resource = self._get_resource()
+            if resource is None:
+                return False
+            logging.info("Existing resource detected. Starting Update.")
+        except client.exceptions.ApiException as error:
```

Comment:
if resource was None above is that the same case here as the 404 error?

Reply:
also claims "creating a new resource" how is that being handled? I don't see the return value being used here to create resource etc, am I missing that portion?

Reply:
> Contributor

If the resource is not in kubernete, both cases could happen.

Reply:
> also claims "creating a new resource" how is that being handled? I don't see the return value being used here to create resource etc, am I missing that portion?

The actual create happen in <resource_name>_component.py. 
```
def _submit_job_request(self, request: Dict) -> object:

        if self.resource_upgrade:
            ack_resource = self._get_resource()
            self.initial_status = ack_resource.get("status", None)
            return super()._patch_custom_resource(request)
        else:
            return super()._create_resource(request, 6, 10)
```